[ADD] l10n_* : add all localizations modules for taxes for SYSCOHADA …

gawa-odoo opened the pull request:
…countries

All these countries use the chart of accounts that is defined in l10n_syscohada. The system for the loading of chart template wasn't planned to inherit from another module, and so the loading of the templates had to be adapted.

This commit add the tax report for each localization, and taxes to be able to fill it.

task-2841655

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/136155).

qdp-odoo commented:
You're creating those loc modules because they share the same CoA but don't share their taxes, right? Hence the diff in [account/models/chart_template.py](https://github.com/odoo/odoo/pull/136155/files#diff-f4c0e9f9d6736a6c5313479e63a48eee140d0e3249932cdcf82d69d0bb7865b1)

Well, that should be made clear, explicit in the commit message. :) Please improve it

gawa-odoo commented:
The task is not yet ready so I did plan on making better message when it really was.
I would be ashamed to try to merge this kind of commit message. :see_no_evil:

Levizar commented on addons/l10n_bf/i18n/fr.po:
```diff
@@ -0,0 +1,223 @@
+#. module: l10n_bf
+#: model:account.report.column,name:l10n_bf.account_tax_report_bf_balance
+msgid "Balance"
+msgstr ""
```
balance

gawa-odoo commented on addons/l10n_cg/data/template/account.tax-cg.csv:
```diff
@@ -0,0 +1,35 @@
+"","","","","","","","","","","tax","invoice","","","","",""
+"","","","","","","","","","","base","refund","-purc_exempt","","","",""
+"","","","","","","","","","","tax","refund","","","","",""
+"tva_sale_18","18%","","True","","18.0","percent","none","tax_group_18","","base","invoice","+base_18","","","",""
+"","","","","","","","","","","tax","invoice","+tax_18","pcg_4431","","",""
+"","","","","","","","","","","base","refund","-base_18","","","",""
+"","","","","","","","","","","tax","refund","-tax_18","pcg_4431","","",""
+"tva_purchase_18","18%","","True","","18.0","percent","none","tax_group_18","","base","invoice","+purc_base","","","",""
+"","","","","","","","","","","tax","invoice","+purc_tax","pcg_4452","","",""
+"","","","","","","","","","","base","refund","-purc_base","","","",""
+"","","","","","","","","","","tax","refund","-purc_tax","pcg_4452","","",""
```
And I put them inactive (to not be selectable)

gawa-odoo commented on addons/l10n_cm/data/template/account.tax-cm.csv:
```diff
@@ -0,0 +1,29 @@
+"","","","","","","","","","tax","invoice","+CM_10_tax","pcg_4431","","",""
+"","","","","","","","","","base","refund","-CM_10_base","","","",""
+"","","","","","","","","","tax","refund","-CM_10_tax","pcg_4431","","",""
+"tva_purchase_good_19_25","19.25%","19.25% goods","True","","19.25","percent","purchase","tax_group_19_25","base","invoice","","","","19.25% Produits et services sauf assets",""
```
I've no idea why that's what I put there. It should have been `achats locaux` which is goods in comparison with `services locaux`

gawa-odoo commented on addons/l10n_td/data/template/account.tax-td.csv:
```diff
@@ -0,0 +1,33 @@
+"","","","","","","","","tax","invoice","","","","",""
+"","","","","","","","","base","refund","","-export","","",""
+"","","","","","","","","tax","refund","","","","",""
+"tva_import_0","0% EX","0% (import)","","0.0","","purchase","tax_group_0","base","invoice","","+import_base","","","0% (importation)"
```
That's not the convention. EX for import too

gawa-odoo commented on addons/l10n_bf/data/account_tax_report_data.xml:
```diff
@@ -0,0 +1,424 @@
+        </field>
+        <field name="line_ids">
+            <record id="account_tax_report_line_bf_turnover" model="account.report.line">
+                <field name="name">III. Global Turnover Without VAT</field>
```
Nope. Période d'imposition + Identification du Contribuable

gawa-odoo commented on addons/l10n_bf/data/template/account.tax.group-bf.csv:
```diff
@@ -0,0 +1,4 @@
+"id","name","name@fr","tax_payable_account_id","tax_receivable_account_id"
+"tax_group_0","VAT 0%","T.V.A. 0%","pcg_4431","pcg_4452"
```
We tend not to do so, I think (and the main point is to have accounts)

Levizar replied:
I agree but then on the invoice, we get the recap with some stuff in 0% while it shouldn't be in the 0%.
0% is suppose to be stuff subject to vat that was taxed at 0% while exempted means it can't be taxed.
I don't remember what's our policy about that.

Laura-labe replied:
Hello @Levizar , as far as I know we don't have policy about this topic. It is very country specific and depends on local regulations. I believe there is no specific requirements in Burkina Faso regarding the distinction on the invoice between the exempted operation and the 0% VAT. So I wouldn't change anything.

Levizar commented on addons/l10n_km/data/template/account.tax-km.csv:
```diff
@@ -0,0 +1,33 @@
+"id","name","description","invoice_label","amount","amount_type","type_tax_use","tax_group_id","repartition_line_ids/repartition_type","repartition_line_ids/document_type","repartition_line_ids/account_id","repartition_line_ids/tag_ids","repartition_line_ids/factor_percent","name@fr","description@fr"
+"tva_sale_25","25%","","","25.0","percent","sale","tax_group_25","base","invoice","","+base_25","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+tax_25","","",""
+"","","","","","","","","base","refund","","-base_25","","",""
+"","","","","","","","","tax","refund","pcg_4431","-tax_25","","",""
+"tva_sale_10","10%","","","10.0","percent","sale","tax_group_10","base","invoice","","+base_10","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+tax_10","","",""
+"","","","","","","","","base","refund","","-base_10","","",""
+"","","","","","","","","tax","refund","pcg_4431","-tax_10","","",""
+"tva_sale_7_5","7.5%","","","7.5","percent","sale","tax_group_7_5","base","invoice","","+base_7_5","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+tax_7_5","","",""
+"","","","","","","","","base","refund","","-base_7_5","","",""
+"","","","","","","","","tax","refund","pcg_4431","-tax_7_5","","",""
+"tva_sale_5","5%","","","5.0","percent","sale","tax_group_5","base","invoice","","+base_5","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+tax_5","","",""
+"","","","","","","","","base","refund","","-base_5","","",""
+"","","","","","","","","tax","refund","pcg_4431","-tax_5","","",""
+"tva_sale_3","3%","","","3.0","percent","sale","tax_group_3","base","invoice","","+base_3","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+tax_3","","",""
+"","","","","","","","","base","refund","","-base_3","","",""
+"","","","","","","","","tax","refund","pcg_4431","-tax_3","","",""
```
very weird!

gawa-odoo commented on addons/l10n_bf/data/template/account.tax-bf.csv:
```diff
@@ -0,0 +1,39 @@
+"id","name","description","invoice_label","amount","amount_type","type_tax_use","tax_group_id","repartition_line_ids/repartition_type","repartition_line_ids/document_type","repartition_line_ids/account_id","repartition_line_ids/tag_ids","repartition_line_ids/factor_percent","name@fr","description@fr"
+"tva_sale_18","18%","","","18.0","percent","sale","tax_group_18","base","invoice","","+BF_01||+BF_15_base","","",""
+"","","","","","","","","tax","invoice","pcg_4431","+BF_15_tax","","",""
+"","","","","","","","","base","refund","","-BF_01||-BF_15_base","","",""
+"","","","","","","","","tax","refund","pcg_4431","-BF_15_tax","","",""
```
Indeed

Levizar reviewed (approved):
@robodoo r+

I discussed with LABE. Most of the work seems to be correct.
As this task is blocking another let's merge it and I will continue to check for issue next week.

If I find some, we can make some more pr to correct them.

robodoo commented:
@gawa-odoo @Levizar staging failed: ci/l10n on b83e5de9cf508d05523423a7870b6bc9fd8b70cb (view more at https://runbot.odoo.com/runbot/build/61567991)

gawa-odoo commented:
So, removed all modifs I had in chart_template (except removing the specific logic for syscohada).
Added a method `_get_{code}_account_account` for all localizations

Levizar reviewed (approved):
@robodoo r+
